include/functions.h

```diff
@@ -1561,19 +1561,19 @@ void ListAlloc_Free(ListAlloc* this, void* data);
 void ListAlloc_FreeAll(ListAlloc* this);
 void Main_LogSystemHeap();
 void Main(void*);
-// OSMesgQueue* PadMgr_LockGetControllerQueue(PadMgr* padmgr);
-// void PadMgr_UnlockReleaseControllerQueue(PadMgr* padmgr, OSMesgQueue* ctrlrqueue);
-// void PadMgr_Lock2(PadMgr* padmgr);
-// void PadMgr_Unlock2(PadMgr* padmgr)
-// ? func_800C740C(?);
-// ? func_800C7818(?);
-// ? func_800C7928(?);
-// ? func_800C7948(?);
-// ? func_800C7974(?);
-// ? func_800C7C14(?);
-// ? func_800C7DD0(?);
-void func_800C7E08(Input*, u32);
-// void PadMgr_Init(PadMgr* padmgr, OSMesg mesg, UNK_TYPE arg2, OSId id, OSPri priority, void* stack);
+// OSMesgQueue* PadMgr_LockSerialMesgQueue(PadMgr* padmgr);
+// void PadMgr_UnlockSerialMesgQueue(PadMgr* padmgr, OSMesgQueue* ctrlrqueue);
+// void PadMgr_LockPadData(PadMgr* padmgr);
+// void PadMgr_UnlockPadData(PadMgr* padmgr);
+// void PadMgr_RumbleControl(PadMgr* padmgr);
+// void PadMgr_RumbleStop(PadMgr* padmgr);
+// void PadMgr_RumbleReset(PadMgr* padmgr);
+// void PadMgr_RumbleSet(PadMgr* padmgr, u8* ctrlr_rumbles);
+// void PadMgr_ProcessInputs(PadMgr* padmgr);
+// void PadMgr_HandleRetraceMsg(PadMgr* padmgr);
+// void PadMgr_HandlePreNMI(PadMgr* padmgr);
+//void PadMgr_RequestPadData(PadMgr* padmgr, Input* inputs, s32 mode);
+//void PadMgr_Init(PadMgr* padmgr, OSMesg mesg, UNK_TYPE arg2, OSId id, OSPri priority, void* stack);
```

Comment:
Is there a reason to keep all these prototypes under comments? Fyi, if they aren't used in other files, they can be removed from the global functions.h. If they are used in other files however, they should ideally have all their correct prototypes available here for the compiler to use.

Reply:
Yes--the types that these functions use are not defined before functions.h is included. If we e.g. included padmgr.h in z64.h, it would be included before the types used in it are defined, so that's not an option either. We'd have to go with some forward-declaration scheme, or else ideally get rid of functions.h and variables.h and create proper include files throughout the repo. Not sure what the best solution is here.

Reply:
The best solution for now is most likely to remove padmgr.h and have the types in z64.h, like we do with everything else while we're actively decompiling things. It was too premature for the padmgr.h file to be added, but I think that was missed at some point in a previous PR.
Eventually we will split up functions.h, variables.h and all the other global headers like z64.h, but right now that would only slow us down more than anything else.

Reply:
Got it. Changed.